bin/shellcheck.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# lint all shell scripts with ShellCheck
+# FUTUREWORK: Fix issues of the explicitly (no globbing) excluded files.
+
+mapfile -t SHELL_FILES_TO_LINT < <(
+    git ls-files |
+        grep "\.sh$" |
+        grep -v "ansible/files/registry/images.sh" |
```

Comment:
Can we use regular expressions(grep doesn't support wildcards, maybe some other solution) here to get all shell files, so that we do not need to update this file in the future in case we add more shell files to our code?

Reply:
This acts like a filter; `grep -v` means *negate result*.

So, the pipeline is built like this:
- `git ls-files` -> list all file names
- `grep "\.sh$"` -> reduce to file name that have the suffix `.sh`
- `grep -v "ansible/files/registry/images.sh"` -> take this one out

The last filters are meant to be removed in future (https://wearezeta.atlassian.net/browse/SQPIT-1051).

*Then, why do these filters exist, anyways?*
There is no type system for shell scripts. And, we have (almost) no tests for them. Thus, every change could break a script. Even worse, it could break a script in a corner that is not often used (e.g. some rarely executed `else` branch). Worst case would be that this would be found in some stressful P1 / fire :fire: nightshift. 
With this conservative approach, I'd like to ensure that we first only apply *ShellCheck* to new scripts and to those that were already free of issues. (I.e. ensure things get better and not worse.) Afterwards, we can fix the issues file by file and apply manual testing to ensure everything works as expected.